### ISSUE TITLE
Symbol starting with _ and listed in __all__ still triggers reportPrivateUsage

### ISSUE THREAD

**Issue by twoertwein** [microsoft/pyright#5900]
**Describe the bug**
My understanding of https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface is that `__all__` affects `from test import *` and is a way for type checkers to know that private-looking symbols (starting with _) are meant to be public:

> All symbols included in the __all__ list are considered public even if the other rules above would otherwise indicate that they were private. For example, this allows symbols whose names begin with an underscore to be included in the interface.

**Code or Screenshots**

```python
# test/py.typed
# test/__init__.py
x = 5
y = 6
_x = 1
_y = 1
__all__ = ['y', '_y']  # mark _y as public

# foo.py
# pyright: strict
import test

print(test.x)
print(test._x)  # reportPrivateUsage
print(test.y)
print(test._y)  # reportPrivateUsage <- shouldn't this be public?
```

**Comment by erictraut** [comment]
I'm not able to repro the problem. I manually created the `test` directory within my `site-packages` directory and created `foo.py` within my local project. Here's what I see.

![image](https://github.com/microsoft/pyright/assets/7040122/08b04396-9982-4a0c-88c8-56e860cf1312)

Are there any other steps I need to take to repro the problem?

**Comment by twoertwein** [comment]
> I'm not able to repro the problem. I manually created the `test` directory within my `site-packages` directory and created `foo.py` within my local project. Here's what I see.

I did not test it in the `site-packages` folder but directly in the current folder.
